cura/Arranging/ArrangeObjectsJob.py

```diff
@@ -35,11 +35,11 @@ def run(self):
             Logger.logException("e", "Unable to arrange the objects on the buildplate. The arrange algorithm has crashed.")
 
         status_message.hide()
-        if found_solution_for_all is not None and not found_solution_for_all:
+        if not found_solution_for_all and found_solution_for_all is not None:
```

Comment:
I agree that further simplification would be ideal.

The original code checks for `if not found_solution_for_all:`
not for `if found_solution_for_all:` like in your comment.
Then the `no_full_solution_message` gets created.

I don’t think a bug would be created by the change I propose since we have an `and` statement so both sides have to be evaluated in any case…

None is falsy so we could just do:
`if not found_solution_for_all:`
I wasn’t sure if there was not an edge case which needed to distinguish False from None … Hence I did not simplify further and mirrored the original.

I think combining with the try block above might be a nice solution.
I’ll open a separate pull request so that you can evaluate this alternative approach...

Reply:
See #12867 for alternative approach.

Reply:
> I don’t think a bug would be created by the change I propose since we have an and statement so both sides have to be evaluated in any case…

Not so. The infix `and`, like in many languages, will shortcut the evaluation if the answer is already known with the first operand. Just like with `or`, where the second statement won't evaluate if the first is already `True`, and won't evaluate when the first statement is already `False`.

As an example, run this simple program:
```
def a() -> bool:
	print("A")
	return False

def b() -> bool:
	print("B")
	return True

if a() and b():
	print("this won't print in any case")
else:
	print("this will print")
```

Reply:
Yes of course there is short circuit evaluation.
I misunderstood your concern.
I though you were worried about the execution of the code block underneath the if statement not the if statement itself…
Since it is the if clause not the else clause both sides have to be evaluated for it to run (there can be no short circuit).

With regards to the if statement itself. I don’t think the change would introduce a bug because in this case we are dealing with variables not functions like in your example...

```
var1 = None

def do_something_and_return_false():
	print("doing something…")
	return False

var2 = do_something_and_return_false()  # this is when “doing something” will be printed

if var2 and var1:
	print("this won't print in any case")
else:
  print("Inside if block")
  print("this will print - var2 already set so not doing anything")

```

will evaluate to:

```
doing something…
Inside if block
this will print - var2 already set so not doing anything
```

Reply:
This https://github.com/Ultimaker/Cura/pull/13060 is much cleaner and simpler.
I don't think **found_solution_for_all** should be _None_ in the first place.